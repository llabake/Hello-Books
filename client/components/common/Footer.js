import React from 'react';
import { Link } from 'react-router-dom';

const Footer = () => {
  const presentYear = new Date().getFullYear();
  return (
    <footer className="page-footer center">
      <div className="container">
        <div className="row">
          <div className="col s12 m4 l3">
            <h5 className="white-text">My Account</h5>
            <ul className="navigation">
              <li><Link to="/" className="grey-text text-lighten-4">Home</Link></li>
              <li><Link to="/profile" className="white-text text-lighten-4">Profile</Link></li>
              <li><Link to="/favorite" className="grey-text text-lighten-4">Favorite</Link></li>
            </ul>
          </div>
          <div className="col s12 m4 l6">
            <h5 className="white-text">Hello Books</h5>
            <p className="white-text text-lighten-4">Stay Connected. Join our over 5 million users today</p>
          </div>
          <div className="col s12 m4 l3" >
            <h5 className="white-text">Be Social</h5>
            <ul className="list-inline">
              <li>
                <a href="https://facebook.com/hellobooksapp" className="social-btn" ><i className="fa fa-facebook-square fa-2x icon-style" /></a>
              </li>
              <li><a href="https://instagram.com/hellobooksapp" className="social-btn"><i className="fa fa-instagram fa-2x icon-style" /></a></li>
              <li><a href="https://twitter.com/hellobooksapp" className="social-btn" ><i className="fa fa-twitter-square fa-2x icon-style" /></a></li>
              <li><a href="https://pinterest.com/hellobooksapp" className="social-btn"><i className="fa fa-pinterest-square fa-2x icon-style" /></a></li>
            </ul>
          </div>
        </div>
      </div>
      <div className="footer-copyright center-align">
        <div className="container">
          Â© {presentYear} Happy Girl World
      </div>
      </div>
    </footer>
  );
};

export default Footer;

