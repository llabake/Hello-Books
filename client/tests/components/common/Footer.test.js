import React from 'react';

import Footer from '../../../components/common/Footer'

const wrapper = shallow(<Footer />);

test('The Footer component mounts correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Footer renders with a footer element', () => {
  expect(wrapper.find('footer').length).toBe(1);
});

test('Footer should have an unordered list element', () => {
  expect(wrapper.find('ul').length).toBe(2);
});

test('Footer list items has seven child elements', () => {
  expect(wrapper.find('ul').children().length).toBe(7);
});

test('Footer links should be rendered', () => {
  expect(wrapper.find('Link').length).toBe(3);

})
test('Footer displays the current year', () => {
  const currentYear = (new Date).getFullYear()

  const copyrightText = wrapper.find('.footer-copyright').text()
  expect(copyrightText).toMatch(String(currentYear))
});

// import React from 'react';
// import { MemoryRouter } from 'react-router-dom';

// import Footer from '../../../components/common/Footer'

// test(' the Footer component mounts correctly', () => {
//   const wrapper = mount(
//     <MemoryRouter>
//       <Footer />
//     </MemoryRouter>
//   );

//   expect(wrapper).toMatchSnapshot();
// })

// test('Footer displays the current year', () => {
//   const currentYear = (new Date).getFullYear()

  
//   const wrapper = shallow(<Footer />);
//   const copyrightText = wrapper.find('.footer-copyright').text()
//   expect(copyrightText).toMatch(String(currentYear))
// })

