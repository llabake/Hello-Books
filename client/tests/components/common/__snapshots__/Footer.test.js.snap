// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Footer component mounts correctly 1`] = `
<footer
  className="page-footer center"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col s12 m4 l3"
      >
        <h5
          className="white-text"
        >
          My Account
        </h5>
        <ul
          className="navigation"
        >
          <li>
            <Link
              className="grey-text text-lighten-4"
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>
          <li>
            <Link
              className="white-text text-lighten-4"
              replace={false}
              to="/profile"
            >
              Profile
            </Link>
          </li>
          <li>
            <Link
              className="grey-text text-lighten-4"
              replace={false}
              to="/favorite"
            >
              Favorite
            </Link>
          </li>
        </ul>
      </div>
      <div
        className="col s12 m4 l6"
      >
        <h5
          className="white-text"
        >
          Hello Books
        </h5>
        <p
          className="white-text text-lighten-4"
        >
          Stay Connected. Join our over 5 million users today
        </p>
      </div>
      <div
        className="col s12 m4 l3"
      >
        <h5
          className="white-text"
        >
          Be Social
        </h5>
        <ul
          className="list-inline"
        >
          <li>
            <a
              className="social-btn"
              href="https://facebook.com/hellobooksapp"
            >
              <i
                className="fa fa-facebook-square fa-2x icon-style"
              />
            </a>
          </li>
          <li>
            <a
              className="social-btn"
              href="https://instagram.com/hellobooksapp"
            >
              <i
                className="fa fa-instagram fa-2x icon-style"
              />
            </a>
          </li>
          <li>
            <a
              className="social-btn"
              href="https://twitter.com/hellobooksapp"
            >
              <i
                className="fa fa-twitter-square fa-2x icon-style"
              />
            </a>
          </li>
          <li>
            <a
              className="social-btn"
              href="https://pinterest.com/hellobooksapp"
            >
              <i
                className="fa fa-pinterest-square fa-2x icon-style"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="footer-copyright center-align"
  >
    <div
      className="container"
    >
      Â© 
      2018
       Happy Girl World
    </div>
  </div>
</footer>
`;
